search-service/src/test/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeServiceTests.kt

```diff
@@ -282,6 +283,164 @@ class TemporalEntityAttributeServiceTests : WithTimescaleContainer, WithKafkaCon
         }
     }
 
+    @Test
+    fun `it should merge an entity attribute`() = runTest {
+        val rawEntity = loadSampleData()
+
+        coEvery { attributeInstanceService.create(any()) } returns Unit.right()
+
+        temporalEntityAttributeService.createEntityTemporalReferences(rawEntity, listOf(APIC_COMPOUND_CONTEXT))
+            .shouldSucceed()
+
+        val temporalEntityAttribute = temporalEntityAttributeService.getForEntityAndAttribute(
+            beehiveTestCId,
+            INCOMING_PROPERTY
+        ).shouldSucceedAndResult()
+
+        val mergedAt = ngsiLdDateTime()
+        val observedAt = ngsiLdDateTime()
+        val newProperty = loadSampleData("fragments/beehive_mergeAttribute.json")
+        val expandedAttribute = JsonLdUtils.expandAttribute(newProperty, listOf(APIC_COMPOUND_CONTEXT))
+        temporalEntityAttributeService.mergeAttribute(
+            temporalEntityAttribute,
+            INCOMING_PROPERTY,
+            AttributeMetadata(
+                null,
+                "It's a string now",
+                null,
+                TemporalEntityAttribute.AttributeValueType.STRING,
+                null,
+                TemporalEntityAttribute.AttributeType.Property,
+                ZonedDateTime.parse("2022-12-24T14:01:22.066Z")
+            ),
+            mergedAt,
+            observedAt,
+            expandedAttribute.second[0],
+            null
+        )
+
+        temporalEntityAttributeService.getForEntityAndAttribute(
+            beehiveTestCId,
+            INCOMING_PROPERTY
+        ).shouldSucceedWith {
+            it.attributeType == TemporalEntityAttribute.AttributeType.Property &&
+                it.attributeValueType == TemporalEntityAttribute.AttributeValueType.STRING &&
+                it.entityId == beehiveTestCId &&
+                it.payload.asString() == serializeObject(expandedAttribute) &&
```

Comment:
I'm a bit surprised with this check. Since the original entity has a `incoming` property, the resulting payload should be the merge of the original and the one you are using in the merge operation (and not the `expandedAttribute`)

Reply:
this is because the payload that we retrieve is that of the `incoming` attribute

Reply:
yes... but I don't see how it answers my point...